src/components/FolderModal.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import PropTypes from 'prop-types';
+import elementStyles from '../styles/isomer-cms/Elements.module.scss';
+import SaveDeleteButtons from './SaveDeleteButtons';
+import FormField from './FormField';
+
+// axios settings
+axios.defaults.withCredentials = true
+
+const FolderModal = ({ displayTitle, displayText, onClose, category, siteName, isCollection }) => {
+  const [newCategoryName, setNewCategoryName] = useState(category)
+
+  const folderNameChangeHandler = (event) => {
+    const { value } = event.target
+    setNewCategoryName(value)
+  }
+
+  const saveHandler = async () => {
+    try {
+      await axios.post(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/${isCollection ? 'collections' : 'resources'}/${category}/rename/${newCategoryName}`)
+      // Refresh page
+      window.location.reload();
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  return (
+    <div className={elementStyles.overlay}>
+      <div className={elementStyles['modal-settings']}>
+        <div className={elementStyles.modalHeader}>
+          <h1>
+            {displayTitle}
+          </h1>
+          <button type="button" onClick={onClose}>
+            <i className="bx bx-x" />
+          </button>
+        </div>
+        <form className={elementStyles.modalContent}>
+          <FormField
+            title={displayText}
+            id="newCategoryName"
+            value={newCategoryName}
+            onFieldChange={folderNameChangeHandler}
+          />
+          <div className={elementStyles.modalButtons}>
+            <SaveDeleteButtons
+              isDisabled={false}
+              hasDeleteButton={false}
+              saveCallback={saveHandler}
+            />
+          </div>
+        </form>
+      </div>
+    </div>
+  )
+};
+
+FolderModal.propTypes = {
+  displayTitle: PropTypes.string.isRequired,
+  displayText: PropTypes.string.isRequired,
+  onProceed: PropTypes.func.isRequired,
+  onCancel: PropTypes.func.isRequired,
```

Comment:
These props are marked as `isRequired` but aren't defined on the component

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/245/commits/697b4c7cbed36050b2a91809324cf3fc3fb5200b